packages/language-server/test/ui5-model-spec.ts

```diff
@@ -29,5 +41,110 @@ describe("the UI5 language assistant ui5 model", () => {
       "sap.ca.ui.charts.ChartSelectionMode"
     );
-  }).timeout(5000); // The default timeout is 2000ms and this test takes ~3000ms
+  }
+
+  it("will get UI5 semantic model", async () => {
+    const ui5Model = await getSemanticModel(undefined);
+    assertSemanticModel(ui5Model);
+  }).timeout(GET_MODEL_TIMEOUT);
+
+  it("doesn't fail if a file cannot be fetched", async () => {
+    const ui5Model = await getSemanticModelWithFetcher(async (url: string) => {
+      return {
+        ok: false,
+        json: (): never => {
+          throw new Error(`Cannot read from ${url}`);
+        }
+      };
+    }, undefined);
+    expect(ui5Model).to.exist;
+  });
+
+  describe("with cache", () => {
+    describe("cache in temp dir", () => {
+      let cachePath: string;
+      beforeEach(async () => {
+        ({ path: cachePath } = await tempDir());
+      });
+
+      afterEach(async () => {
+        // The temp folder will contain files at the end so we remove it
+        // with rimraf instead of calling cleanup()
+        rimrafSync(cachePath);
+      });
+
+      it("caches the model the first time getSemanticModel is called", async () => {
+        const ui5Model = await getSemanticModel(cachePath);
+        assertSemanticModel(ui5Model);
+
+        // Check the files were created in the folder
+        const files = await readdir(cachePath);
+        expect(files.length).to.be.greaterThan(0);
+
+        // Call getSemanticModel again with the same path and check it doesn't try to read from the URL
+        let fetcherCalled = false;
+        const ui5ModelFromCache = await getSemanticModelWithFetcher(
+          (url: string): never => {
+            fetcherCalled = true;
+            throw new Error(
+              `The files should be taken from the cache, got call for ${url}`
+            );
+          },
+          cachePath
+        );
+        expect(fetcherCalled).to.be.false;
+        // Make sure it's not the model itself that is cached
+        expect(ui5ModelFromCache).to.not.equal(ui5Model);
+        assertSemanticModel(ui5ModelFromCache);
+      }).timeout(GET_MODEL_TIMEOUT);
+
+      it("doesn't fail when file cannot be written to the cache", async () => {
+        // Create a folder with the file name so the file will not be written
+        await mkdirs(
+          getCacheFilePath(getCacheFolder(cachePath, VERSION), "sap.m")
+        );
+        const ui5Model = await getSemanticModel(cachePath);
+        expect(ui5Model).to.exist;
+      }).timeout(GET_MODEL_TIMEOUT);
+
+      it("doesn't fail when file cannot be read from the cache", async () => {
+        // Create a file with non-json content so the file will not be deserialized
+        const cacheFolder = getCacheFolder(cachePath, VERSION);
+        await mkdirs(cacheFolder);
+        await writeFile(getCacheFilePath(cacheFolder, "sap.m"), "not json");
+        const ui5Model = await getSemanticModel(cachePath);
+        expect(ui5Model).to.exist;
```

Comment:
again assert something from sap.m is actually contained in the model

Reply:
Done